pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -227,24 +229,45 @@ func init() {
 	})
 }
 
+// UUID gets the BIOS UUID via the sys interface.  This UUID is known by vsphere
+func getvmUUID() (string, error) {
+	id, err := ioutil.ReadFile(UUIDPath)
+	if err != nil {
+		return "", fmt.Errorf("error retrieving vm uuid: %s", err)
+	}
+
+	uuidstr := string(id[:])
+
+	// check the uuid starts with "VMware-"
+	if !strings.HasPrefix(uuidstr, UUIDPrefix) {
+		return "", fmt.Errorf("cannot find this VM's UUID")
+	}
+
+	// Strip the prefix, white spaces, and the trailing '\n'
+	uuidstr = strings.Replace(uuidstr[len(UUIDPrefix):(len(uuidstr)-1)], " ", "", -1)
+
+	// need to add dashes, e.g. "564d395e-d807-e18a-cb25-b79f65eb2b9f"
+	uuidstr = fmt.Sprintf("%s-%s-%s-%s", uuidstr[0:8], uuidstr[8:12], uuidstr[12:21], uuidstr[21:])
```

Comment:
Validate the length of `uuidstr` in case it's shorter than expected by accident.